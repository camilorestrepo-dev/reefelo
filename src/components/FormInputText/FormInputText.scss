@import "../../styles/shared.scss";

label.form-input {
  display: flex;
  flex-direction: column;
  width: 100%;
  gap: $spacing-4;
  margin-bottom: $spacing-16;

  &.input-error {
    input {
      border-color: $error-color;
    }
  }

  input {
    padding: $spacing-4 $spacing-8;
    font-size: large;
    border-radius: $radius-5;
    border: 1px solid $grey;
  }

  .error-message {
    color: $error-color;
    font-size: smaller;
  }
}