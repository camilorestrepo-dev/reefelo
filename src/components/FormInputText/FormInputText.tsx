import { ChangeEventHandler, HTMLInputTypeAttribute } from "react";
import "./FormInputText.scss";

type FormInputTextProps = {
  label: string;
  name: string;
  placeholder?: string;
  type?: HTMLInputTypeAttribute;
  defaultValue: string | number | readonly string[];
  onChange?: ChangeEventHandler<HTMLInputElement>;
};

const FormInputText: React.FC<FormInputTextProps> = ({
  label,
  name,
  placeholder,
  type = "text",
  defaultValue,
  onChange,
}: FormInputTextProps) => {
  return (
    <label className="form-input">
      <span>{label}</span>
      <input
        type={type}
        name={name}
        placeholder={placeholder}
        defaultValue={defaultValue}
        onChange={onChange}
      />
    </label>
  );
};

export default FormInputText;
