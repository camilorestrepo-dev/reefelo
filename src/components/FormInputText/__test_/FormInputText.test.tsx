import { render } from "@testing-library/react";
import FormInputText from "../FormInputText";

test("reder corectly", () => {
  const component = render(
    <FormInputText
      defaultValue="hello"
      label="Description"
      name="desc"
      placeholder="Type description"
    />,
  );
  expect(component.container).toMatchSnapshot();
});
