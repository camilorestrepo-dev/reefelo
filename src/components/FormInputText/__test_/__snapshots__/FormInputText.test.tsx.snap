// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reder corectly 1`] = `
<div>
  <label
    class="form-input"
  >
    <span>
      Description
    </span>
    <input
      name="desc"
      placeholder="Type description"
      type="text"
      value="hello"
    />
  </label>
</div>
`;

exports[`reder corectly with error 1`] = `
<div>
  <label
    class="form-input input-error"
  >
    <span>
      Description
    </span>
    <input
      name="desc"
      placeholder="Type description"
      type="text"
      value="hello"
    />
    <span
      class="error-message"
    >
      Required field
    </span>
  </label>
</div>
`;
